external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```
@@ -116,217 +141,57 @@
-            return new Builder<>(this, null, clazz, valueDes, valueDesClazz);
-        }
-
-        /**
-         * Specifying this value deserializer overrides the property value.deserializer.  If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewValueT> Builder<K,NewValueT> setValue(SerializableDeserializer<NewValueT> valueDeserializer) {
-            return new Builder<>(this, keyDes, keyDesClazz, valueDeserializer, null);
+            this.translator = new DefaultRecordTranslator<>();
         }
         
         /**
-         * Specify a class that can be instantiated to create a value.deserializer
-         * This is the same as setting value.deserializer, but overrides it.  If you have
-         * set a custom RecordTranslator before calling this it may result in class cast
-         * exceptions at runtime.
-         */
-        public <NewValueT> Builder<K,NewValueT> setValue(Class<? extends Deserializer<NewValueT>> clazz) {
-            return new Builder<>(this, keyDes, keyDesClazz, null, clazz);
-        }
-        
-        /**
-         * Set a Kafka property config.
+         * Set a Kafka consumer property config. 
          */
         public Builder<K,V> setProp(String key, Object value) {
             kafkaProps.put(key, value);
             return this;
         }
         
         /**
-         * Set multiple Kafka property configs.
+         * Set multiple Kafka consumer property configs.
          */
         public Builder<K,V> setProp(Map<String, Object> props) {
             kafkaProps.putAll(props);
             return this;
         }
         
         /**
-         * Set multiple Kafka property configs.
+         * Set multiple Kafka consumer property configs.
          */
         public Builder<K,V> setProp(Properties props) {
             for (String name: props.stringPropertyNames()) {
```

Nice catch. How about we use putAll to put the properties into kafkaProps instead? I'm not sure that I like the method filtering the input properties, I'd rather throw an error if the key is not a String. At the same time I don't think it should be KafkaSpoutConfig's job to validate KafkaConsumer parameters. The KafkaConsumer will handle throwing the error if the keys are not Strings https://github.com/apache/kafka/blob/efb060c57f05d1d586bb14c016b0187c60f8e994/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L60